components/description-list/description-list-wrapper.js

```diff
@@ -0,0 +1,100 @@
+import { bodyCompactStylize, labelStylize } from '../typography/styles.js';
+import { css, html, LitElement } from 'lit';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+
+export const descriptionListStyles = [
+	labelStylize(css`dt`),
+	bodyCompactStylize(css`dd`),
+	css`
+		:host {
+			--d2l-dt-min-width: min-content;
+			--d2l-dt-max-width: 10rem;
+			--d2l-dd-min-width: 50%;
+		}
+		dl {
+			display: grid;
+			grid-auto-flow: row;
+			align-items: baseline;
+			grid-template-columns: minmax(var(--d2l-dt-min-width), auto) minmax(var(--d2l-dd-min-width), 1fr);
+			gap: 0.5rem 1.5rem;
+		}
+		dt {
+			max-width: var(--d2l-dt-max-width);
+		}
+		dd {
+			margin: 0;
+		}
+		d2l-dl-wrapper[stacked] dl {
+			display: block;
+		}
+		d2l-dl-wrapper[stacked] dt {
+			max-width: unset;
+		}
+		d2l-dl-wrapper[stacked] dd {
+			margin-bottom: 0.5rem;
+		}
+	`
+];
+
+/**
+ * Wraps a native <dl> element, providing styling and resize behavior.
+ * @slot - Content to wrap
+ */
+class DescriptionListWrapper extends LitElement {
+	static get properties() {
+		return {
+			stacked: { type: Boolean, reflect: true },
+			breakpoint: { type: Number, reflect: true },
+		};
+	}
+
+	static get styles() {
+		return css`
+			:host {
+				display: block;
+			}
```

Comment:
Nit: if you modify the `display`, you need to specify the `hidden` behaviour:
https://daylight.d2l.dev/developing/best-practices/styles/#do%3A-style-the-%22display%22-value-of-the-host